gensim/models/coherence_utils.py

```diff
@@ -4,25 +4,539 @@
 # Copyright (C) 2013 Radim Rehurek <radimrehurek@seznam.cz>
 # Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
 
+
 """
-This module contains classes for analyzing the texts of a corpus to accumulate
-statistical information about word occurrences.
+This module contains implementation of the individual components of
+the topic coherence pipeline.
 """
 
-import itertools
-import logging
-import multiprocessing as mp
 import sys
+import logging
+import itertools
 from collections import Counter
-
+import multiprocessing as mp
 import numpy as np
-import scipy.sparse as sps
-from six import iteritems, string_types
-
-from gensim import utils
+from gensim.utils import iter_windows, flatten
 from gensim.models.word2vec import Word2Vec
+from scipy import sparse as sps
+from six import string_types, iteritems
 
 logger = logging.getLogger(__name__)
+EPSILON = 1e-12  # Should be small. Value as suggested in paper.
+
+
+"""
```

Comment:
The presence of these docstrings throughout the module makes me think the existing module structure was actually a pretty useful way to separate these things out. Have you considered adding a `coherence_utils` package with `from <existing_module> import *` instead of throwing them all in the same module? This seems cleaner to me. I think it makes things easier to find and navigate.

Reply:
So, maybe simple moving & renaming `coherence_model->gensim.models.coherence_utils` OR `gensim.models.coherence_inner`is better and you are right (because I don't think that this variant is sufficiently consistent), for this reason, I'm asking your for review, thanks :+1: